src/main/java/org/damap/base/rest/document/service/DocumentService.java

```diff
@@ -0,0 +1,95 @@
+package org.damap.base.rest.document.service;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.ws.rs.WebApplicationException;
+import jakarta.ws.rs.core.StreamingOutput;
+import java.io.File;
+import java.io.FileOutputStream;
+import java.io.IOException;
+import lombok.extern.jbosslog.JBossLog;
+import org.apache.poi.xwpf.usermodel.XWPFDocument;
+import org.damap.base.conversion.ExportTemplateBroker;
+import org.damap.base.conversion.TemplateSelectorServiceImpl;
+import org.damap.base.enums.ETemplateType;
+import org.damap.base.rest.dmp.service.DmpService;
+import org.damap.base.rest.document.dto.MultipartBodyDTO;
+import org.eclipse.microprofile.rest.client.inject.RestClient;
+
+@ApplicationScoped
+@JBossLog
+public class DocumentService {
+
+  @Inject ExportTemplateBroker exportTemplateBroker;
+
+  @Inject TemplateSelectorServiceImpl templateSelectorService;
+
+  @Inject DmpService dmpService;
+
+  @Inject @RestClient GotenbergRestService gotenbergRestService;
+
+  public StreamingOutput getExportDocument(
+      long dmpId, ETemplateType template, boolean download, String filetype) {
+    // Fetch the document based on the template
+    XWPFDocument document =
+        (template != null)
+            ? exportTemplateBroker.exportTemplateByType(dmpId, template)
+            : exportTemplateBroker.exportTemplate(dmpId);
+
+    if (filetype.equals("pdf")) {
+      return getPdfOf(document);
+    } else if (filetype.equals("docx")) {
+      return getWordDocumentOf(document);
+    }
+
+    throw new WebApplicationException("Invalid file type: " + filetype);
+  }
+
+  private StreamingOutput getPdfOf(XWPFDocument xwpfDocument) {
+    File tempFile;
+    try {
+      tempFile = File.createTempFile("dmp", ".docx");
```

Comment:
major: in order to keep possible file clashes and file overrides to a minimum, adding the dmp id and current timestamp might be a good idea.

Reply:
Done